### ISSUE TITLE
Crash on entity wipe

### ISSUE THREAD

**Issue by ScrachStack** [citizenfx/fivem#2344]
### What happened?

```lua
RegisterCommand("entitywipe", function()
    local playerPed = cache.ped
    local playerCoords = GetEntityCoords(playerPed)

    local radius = 20.0 

    local entityPool = GetGamePool('CPed')
    for i = 1, #entityPool do
        local distance = #(GetEntityCoords(entityPool[i]) - playerCoords)

        if distance < radius then
                DeleteEntity(entityPool[i])
            
        end
    end

    RedTrainer.Notify("Entity Wipe", "20.0 Radius Wiped.", 'info')
end, true)
``` this code causes a crash

### Expected result

entity should have wiped

### Reproduction steps

run command above

### Importancy

Crash

### Area(s)

RedM

### Specific version(s)

REDM GAME BUILD 1491 LATEST artifacts AS OF 1/16/2024

### Additional information

![image](https://github.com/citizenfx/fivem/assets/122732007/ebeacfa9-881f-4c3b-ae12-9832e043dee7)
[CfxCrashDump_2024_01_16_18_06_23.zip](https://github.com/citizenfx/fivem/files/13954192/CfxCrashDump_2024_01_16_18_06_23.zip)

**Comment by blattersturm** [comment]
I suppose RedM doesn't have an equivalent to this FiveM patch that prevents deleting the local player ped:

https://github.com/citizenfx/fivem/blob/c18c3214cb5d2fec62cd59e528073442bc063c63/code/components/extra-natives-five/src/EntityDeletionNatives.cpp#L58-L66

**Comment by Disquse** [comment]
> I suppose RedM doesn't have an equivalent to this FiveM patch that prevents deleting the local player ped, as that would crash the game:

But RedM shouldn't crash if you remove your local ped 🤔

**Comment by Zaps6000** [comment]
> > I suppose RedM doesn't have an equivalent to this FiveM patch that prevents deleting the local player ped, as that would crash the game:
> 
> But RedM shouldn't crash if you remove your local player ped 🤔

First time this happened I died and couldn't respawn as my pet wasn't there the 2nd time I joined back and tried it I crashed I posted it above if it helps.

**Comment by ScrachStack** [comment]
..

**Comment by outsider31000** [comment]
I dont seem to crash with this can you perhaps try again @ScrachStack

**Comment by ScrachStack** [comment]
it could have been updated since but i crash

**Comment by ScrachStack** [comment]
there is no code to stop your entity being deleted which breaks the game

**Comment by ScrachStack** [comment]
this is a redm exclusive issue which was supposed to be getting worked on someone was tasked

**Comment by outsider31000** [comment]
yeah it seems that it doesnt check if the entity is a player like it does for fivem just straight deletes it

https://github.com/citizenfx/fivem/blob/c3aa4e62e12b991a63190017d8a423cccd56e1fa/code/components/gta-core-rdr3/src/EntityDeletionHacks.cpp#L311

https://github.com/citizenfx/fivem/blob/c3aa4e62e12b991a63190017d8a423cccd56e1fa/code/components/gta-core-rdr3/src/EntityDeletionHacks.cpp#L78

**Comment by outsider31000** [comment]
thanks for reporting the issue a fix is in place
